File: 728-self-dividing-numbers/728.py
class Solution:
    def isSelfDividing(self, n):
        orig = n
        while n > 0:
            if n % 10 == 0:
                return False
            if (orig % (n % 10)) != 0:
                return False
            n //= 10
        return True

    def selfDividingNumbers(self, left: int, right: int) -> List[int]:
        ans = []
        for i in range(left, right+1):
            if self.isSelfDividing(i):
                ans.append(i)
        return ans

File: 728-self-dividing-numbers/728.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
bool isSelf(int target) {
    int n = target;
    while (n) {
        int x = n % 10;
        if (x == 0)
            return false;
        if (target % x != 0)
            return false;
        n /= 10;
    }
    return true;
}

int* selfDividingNumbers(int left, int right, int* returnSize){
    int *ret = (int*)calloc(sizeof(int), right - left);
    int index = 0;
    for (int i = left; i <= right; ++i) {
        if (isSelf(i))
            ret[index++] = i;
    }
    *returnSize = index;
    return ret;
}
